File: src/main/java/de/gesellix/testutil/ResourceReader.java
package de.gesellix.testutil;

import java.io.File;
import java.net.URISyntaxException;

public class ResourceReader {

  public static File getClasspathResourceAsFile(String classpathResource, Class<?> baseClass) {
    try {
      return new File(baseClass.getResource(classpathResource).toURI());
    }
    catch (URISyntaxException e) {
      throw new RuntimeException(e);
    }
  }
}

File: src/main/kotlin/de/gesellix/testutil/ResourceReader.kt
package de.gesellix.testutil

import java.io.File

class ResourceReader {

    fun getClasspathResourceAsFile(classpathResource: String, baseClass: Class<*>): File {
        val resource = baseClass.getResource(classpathResource)
        return File(resource.toURI())
    }
}
